propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagator/ot_trace/__init__.py

```diff
@@ -0,0 +1,173 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from re import compile as re_compile
+from typing import Iterable, Optional
+
+from opentelemetry.baggage import get_all, set_baggage
+from opentelemetry.context import Context
+from opentelemetry.trace import (
+    INVALID_SPAN_ID,
+    INVALID_TRACE_ID,
+    DefaultSpan,
+    SpanContext,
+    TraceFlags,
+    get_current_span,
+    set_span_in_context,
+)
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+OT_TRACE_ID_HEADER = "ot-tracer-traceid"
+OT_SPAN_ID_HEADER = "ot-tracer-spanid"
+OT_SAMPLED_HEADER = "ot-tracer-sampled"
+OT_BAGGAGE_PREFIX = "ot-baggage-"
+
+_valid_header_name = re_compile(r"^[\w_^`!#$%&'*+.|~]+$")
+_valid_header_value = re_compile(r"^[\t\x20-\x7e\x80-\xff]+$")
+_valid_traceid = re_compile(r"[0-9a-f]{32}|[0-9a-f]{16}")
+_valid_spanid = re_compile(r"[0-9a-f]{16}")
```

Comment:
Sorry, what is the issue here?

Reply:
just a nit, if these are constants consider `_valid_header_name` -> `_VALID_HEADER_NAME`

Reply:
PEP8 says that is how [constants](https://www.python.org/dev/peps/pep-0008/#constants) should be cased. AFAIK, constants are only `True`, `False`, `None`, strings, and numerical values.

Reply:
Some compiled regexes in constant case here as well https://github.com/open-telemetry/opentelemetry-python/blob/0fe1f476a05bfa055a868d8433488a785bb65cdc/opentelemetry-api/src/opentelemetry/util/tracestate.py#L49-L52

I believe the purpose of this casing is to indicate the variable is a constant and shouldn't be modified, which is the intention here. int, float, bool, string, etc. aren't special in python, they are just classes right?

Reply:
> Some compiled regexes in constant case here as well https://github.com/open-telemetry/opentelemetry-python/blob/0fe1f476a05bfa055a868d8433488a785bb65cdc/opentelemetry-api/src/opentelemetry/util/tracestate.py#L49-L52
> 

Yeah..., we should also fix those :sweat_smile: 

> I believe the purpose of this casing is to indicate the variable is a constant and shouldn't be modified, which is the intention here. int, float, bool, string, etc. aren't special in python, they are just classes right?

Hmnot really. `int`, `float`, `bool`, `None`, strings *are* special, because they are actually constants and they are created only once. This can be checked like this:

```python
ocelotl@harrison:~$ python
Python 3.8.3 (default, Jun 29 2020, 18:03:36) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = 1
>>> b = 1
>>> id(a)
93987911005952
>>> id(b)
93987911005952
>>> a is b
True
>>> a = "a"
>>> b = "a"
>>> id(a)
140646739870512
>>> id(b)
140646739870512
>>> a is b
True
>>> a = None
>>> b = None
>>> id(a)
93987910826592
>>> id(b)
93987910826592
>>> a is b
True
>>> 
```

As it can be seen here, any variable assigned to the same `int` will "point" to the same object, `int`s are true constants in that sense, they will not change during the program execution.

The same does not happen for other objects:

```python
ocelotl@harrison:~$ python
Python 3.8.3 (default, Jun 29 2020, 18:03:36) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class Test:
...     pass
... 
>>> a = Test()
>>> b = Test()
>>> id(a)
139882366030992
>>> id(b)
139882366749904
>>> a is b
False
>>> 
```

Now, let's try with compiled regular expressions:

```python
ocelotl@harrison:~$ python
fPython 3.8.3 (default, Jun 29 2020, 18:03:36) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from re import compile as re_compile
>>> a = re_compile(r"abc")
>>> b = re_compile(r"abc")
>>> id(a)
140532296161216
>>> id(b)
140532296161216
>>> a is b
True
>>> 
```

Now, this looks like compiled regular expressions are indeed constants the same as `int`s are, but that is just an illusion caused by an implementation [detail](https://github.com/python/cpython/blob/master/Lib/re.py#L285). The `re` module uses an artificial cache as an optimization to keep there already compiled regular expressions. That cache has a [limit](https://github.com/python/cpython/blob/master/Lib/re.py#L297) so at some moment even compiled regular expressions may be compiled again for the exact same pattern and flags. Now, I am getting just pretty pedantic here with the definition of a constant, :sweat_smile: but I wanted to explain why compiled regular expressions are not constants in the most strict sense.